src/share/poudriere/image.sh

```diff
@@ -411,9 +521,16 @@ if [ -f "${WRKDIR}/world/etc/login.conf.orig" ]; then
 fi
 cap_mkdb ${WRKDIR}/world/etc/login.conf
 
-# Set hostname
+# Always set hostname if it was specified on command line
 if [ -n "${HOSTNAME}" ]; then
-	echo "hostname=${HOSTNAME}" >> ${WRKDIR}/world/etc/rc.conf
+	setimagehostname "${HOSTNAME}"	
+else
+	# If hostname wasn't set on command line, only set
+	# default hostname if hostname isn't specified by
+	# overlay file strucutre.
+	if ! sysrc -n -R "${WRKDIR}/world" hostname; then
+		setimagehostname "poudriere-image"
+	fi
```

Comment:
I think this could be simplified like this to only call `setimagehostname` once.
```
: ${HOSTNAME:=$(sysrc -n -q -R "${WRKDIR}/world" hostname || echo "poudriere-image")}
```
It would use the `-h hostname` value by default, then look at sysrc for the hostname, then fallback to the default.